internal/config/client.go

```diff
@@ -0,0 +1,99 @@
+package config
+
+import (
+	"context"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
+)
+
+// NewCachingClient is an alternative implementation of controller-runtime's
+// default client for manager.Manager.
+// The only difference is that this implementation sets `CacheUnstructured` to `true` to
+// cache unstructured objects.
+func NewCachingClient(cache cache.Cache, config *rest.Config, options client.Options, uncachedObjects ...client.Object) (client.Client, error) {
```

Comment:
Should this be an unexported function now?

Reply:
Yes, it can be unexported now. Just wonder why you want to export/unexport based on the currentusage? To me, it is more natural to export/unexport based on weather the symbol could/should be used outside the package.

Reply:
I generally default to "nothing exported unless needed." It makes it that much easier to understand what the entrypoints to a package are when I'm reading the code.

Reply:
Okey!